File: server/datasets/java_source-code/Multiple-E3/type3_reverse_1_max_subarray.java
public class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int max_sum = nums[n - 1];
        int current_sum = nums[n - 1];
        
        for (int i = n - 2; i >= 0; i--) {
            current_sum = Math.max(nums[i], current_sum + nums[i]);
            max_sum = Math.max(max_sum, current_sum);
        }
        return max_sum;
    }
}

File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_e3/type3_reverse_1_max_subarray.py
def maxSubArray(nums):
    n = len(nums)
    max_sum = nums[n-1]
    current_sum = nums[n-1]
    
    for i in range(n-2, -1, -1):
        current_sum = max(nums[i], current_sum + nums[i])
        max_sum = max(max_sum, current_sum)
        
    return max_sum
